packages/mermaid/src/rendering-util/rendering-elements/shapes/tiltedCylinder.ts

```diff
@@ -84,19 +90,19 @@ export const tiltedCylinder = async (parent: SVGAElement, node: Node) => {
       .insert('path', ':first-child')
       .attr('d', pathData)
       .attr('class', 'basic label-container')
-      .attr('style', cssStyles)
+      .attr('style', handleUndefinedAttr(cssStyles))
       .attr('style', nodeStyles);
-  }
+    cylinder.attr('class', 'basic label-container');
 
-  cylinder.attr('class', 'basic label-container');
+    if (cssStyles) {
+      cylinder.selectAll('path').attr('style', cssStyles);
+    }
 
-  if (cssStyles && node.look !== 'handDrawn') {
-    cylinder.selectAll('path').attr('style', cssStyles);
+    if (nodeStyles) {
+      cylinder.selectAll('path').attr('style', nodeStyles);
+    }
   }
 
-  if (nodeStyles && node.look !== 'handDrawn') {
-    cylinder.selectAll('path').attr('style', nodeStyles);
-  }
```

Comment:
Same issue here.

TypeScript isn't smart enough to understand `cylinder.selectAll` when `cylinder` is `D3Selection<SVGGElement> | D3Selection<SVGPathElement>`.

However, if we move it into the `if (node.look !== 'handDrawn')` block, it knows that `cylinder` is only `D3Selection<SVGPathElement>` and nothing else.

The `if (node.look === 'handDrawn' )` block also already has a `cylinder.attr('class', 'basic label-container');` call, so we don't need to copy it there.